worlds/kh2/__init__.py

```diff
@@ -145,175 +173,258 @@ def create_items(self) -> None:
+            self.sora_ability_dict[random_support_ability] += 1
+
         self.plando_locations = dict()
-        self.hitlist = []
         self.starting_invo_verify()
+
+        for k, v in self.options.CustomItemPoolQuantity.value.items():
+            # kh2's items cannot hold more than a byte
+            if 255 > v > self.item_quantity_dict[k] and k in default_itempool_option.keys():
+                self.item_quantity_dict[k] = v
+            elif 255 <= v:
+                logging.warning(
+                        f"{self.player} has too many {k} in their CustomItemPool setting. Setting to default quantity")
         # Option to turn off Promise Charm Item
-        if not self.multiworld.Promise_Charm[self.player]:
+        if not self.options.Promise_Charm:
             self.item_quantity_dict[ItemName.PromiseCharm] = 0
 
+        if not self.options.AntiForm:
+            self.item_quantity_dict[ItemName.AntiForm] = 0
+
         self.set_excluded_locations()
 
-        if self.multiworld.Goal[self.player] == "lucky_emblem_hunt":
-            self.luckyemblemamount = self.multiworld.LuckyEmblemsAmount[self.player].value
-            self.luckyemblemrequired = self.multiworld.LuckyEmblemsRequired[self.player].value
+        if self.options.Goal not in ["hitlist", "three_proofs"]:
+            self.lucky_emblem_amount = self.options.LuckyEmblemsAmount.value
+            self.lucky_emblem_required = self.options.LuckyEmblemsRequired.value
             self.emblem_verify()
 
         # hitlist
-        elif self.multiworld.Goal[self.player] == "hitlist":
-            self.RandomSuperBoss.extend(exclusion_table["Hitlist"])
-            self.BountiesAmount = self.multiworld.BountyAmount[self.player].value
-            self.BountiesRequired = self.multiworld.BountyRequired[self.player].value
+        if self.options.Goal not in ["lucky_emblem_hunt", "three_proofs"]:
+            self.random_super_boss_list.extend(exclusion_table["Hitlist"])
+            self.bounties_amount = self.options.BountyAmount.value
+            self.bounties_required = self.options.BountyRequired.value
 
             self.hitlist_verify()
 
-            for bounty in range(self.BountiesAmount):
-                randomBoss = self.multiworld.per_slot_randoms[self.player].choice(self.RandomSuperBoss)
-                self.plando_locations[randomBoss] = ItemName.Bounty
-                self.hitlist.append(self.location_name_to_id[randomBoss])
-                self.RandomSuperBoss.remove(randomBoss)
-                self.totalLocations -= 1
-
-        self.donald_fill()
-        self.goofy_fill()
-        self.keyblade_fill()
+            prio_hitlist = [location for location in self.multiworld.priority_locations[self.player].value if
+                            location in self.random_super_boss_list]
+            for bounty in range(self.options.BountyAmount.value):
+                if prio_hitlist:
+                    random_boss = self.random.choice(prio_hitlist)
```

Comment:
`random_boss = prio_hitlist.pop(self.random.randrange(len(prio_hitlist)))` should be about the same speed and a bit less code. How good it looks depends on the reader though, so up to your discretion.

Reply:
I think my code just reads better but this is cleaner in a line sense for sure